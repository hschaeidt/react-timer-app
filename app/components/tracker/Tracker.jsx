import React from 'react';
import moment from 'moment';
import TrackFilter from './TrackFilter';
import TrackerList from './TrackerList';
import TrackForm from './TrackForm';
import tracksData from '../../fixtures/tracks.json';

export default class Tracker extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      tracks: tracksData,
      filter: {
        showCompleted: false,
      },
    };

    this.handleFilter = this.handleFilter.bind(this);
    this.handleAddTrack = this.handleAddTrack.bind(this);
  }

  handleFilter(showCompleted) {
    this.setState({
      filter: {
        showCompleted,
      },
    });
  }

  handleAddTrack(track) {
    const tracks = [
      ...this.state.tracks,
      {
        date: track,
        ranges: [{
          from: moment().toISOString(),
          to: moment().add(1, 'hour').toISOString(),
        }],
      },
    ];

    this.setState({
      tracks,
    });
  }

  render() {
    const { tracks } = this.state;

    return (
      <div>
        <h1 className="page-title">Time Tracker</h1>
        <TrackFilter onFilter={this.handleFilter} />
        <TrackerList tracks={tracks} />
        <TrackForm onAddTrack={this.handleAddTrack} />
      </div>
    );
  }
}
