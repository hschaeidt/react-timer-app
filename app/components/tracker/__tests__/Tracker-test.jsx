import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Tracker from '../Tracker';

describe('<Tracker />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Tracker />);
    expect(tree).toMatchSnapshot();
  });

  it('should change the filter status', () => {
    const instance = shallow(<Tracker />).instance();
    expect(instance.state.filter.showCompleted).toBe(false);
    instance.handleFilter(true);
    expect(instance.state.filter.showCompleted).toBe(true);
  });

  it('should add new tracks to the list', () => {
    const instance = shallow(<Tracker />).instance();
    expect(instance.state.tracks.length).toBe(2);
    instance.handleAddTrack('2015-01-01');
    expect(instance.state.tracks.length).toBe(3);
  });
});
