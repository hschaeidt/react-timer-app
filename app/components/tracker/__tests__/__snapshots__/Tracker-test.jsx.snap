exports[`<Tracker /> renders correctly 1`] = `
<div>
  <h1
    className="page-title">
    Time Tracker
  </h1>
  <div>
    <label
      htmlFor="input-show-completed">
      <input
        id="input-show-completed"
        onChange={[Function]}
        type="checkbox" />
      Show completed days
    </label>
  </div>
  <div>
    <div>
      Friday, December 2nd 2016
      <div>
        Time ranges:
        <div>
          from 
          8:57 AM
           to 
          1:01 PM
        </div>
        <div>
          from 
          2:00 PM
           to 
          6:00 PM
        </div>
      </div>
      <div>
        Total time tracked: 
        8
         hours 
        4
         minutes
      </div>
    </div>
    <div>
      Saturday, December 3rd 2016
      <div>
        Time ranges:
        <div>
          from 
          10:00 AM
           to 
          1:00 PM
        </div>
        <div>
          from 
          2:00 PM
           to 
          7:00 PM
        </div>
      </div>
      <div>
        Total time tracked: 
        8
         hours 
        0
         minutes
      </div>
    </div>
  </div>
  <form
    onSubmit={[Function]}>
    <input
      onChange={[Function]}
      type="date"
      value="2016-12-04" />
    <button
      className="button expanded">
      Check-in
    </button>
  </form>
</div>
`;
